runtime/jvmti/suspendhelper.cpp

```diff
@@ -45,7 +45,15 @@ suspendThread(J9VMThread *currentThread, jthread thread, UDATA allowNull, UDATA
 				if (currentThread == targetThread) {
 					*currentThreadSuspended = TRUE;
 				} else {
+					/* It is a well known problem that suspending a Java thread may cause deadlocks in the application if a resource is 
+					 * held while suspended. To prevent this happening in the Thread JCL library, check that the Object java.lang.Thread.lock 
+					 * has been released before suspending. To make sure of this, acquire the lock while halt is in progress.
+					 */
+					j9object_t threadLock = J9VMJAVALANGTHREAD_LOCK(targetThread, targetThread->threadObject);
```

Comment:
Is the intention to match the java implementation of the `Thread.suspend` method by acquiring the lock for other threads before suspending?
https://github.com/eclipse/openj9/blob/ea7b2e9783f21f186eb7ac087af77d74dc5a78ad/jcl/src/java.base/share/classes/java/lang/Thread.java#L1280-L1284

Making this match the Java implementation makes sense to me but I don't understand the comment above

Reply:
I suspect the lock here is to prevent the thread from dying while we attempt to suspend it. JVMTI uses a different mechanism (vmThreadListMutex and inspectorCount).